src/main/scala/com/wanari/tutelar/core/AuthService.scala

```diff
@@ -26,6 +26,7 @@ trait AuthService[F[_]] extends Initable[F] {
   ): ErrorOr[F, (Account, JsObject)]
   def unlink(userId: String, authType: String)(implicit ctx: LogContext): ErrorOr[F, Unit]
   def refreshToken(token: LongTermToken)(implicit ctx: LogContext): ErrorOr[F, TokenData]
+  def findAccessToken(userId: String, authType: String)(implicit ctx: LogContext): ErrorOr[F, String]
```

Comment:
it is an unnecessary function! you should use the `def findCustomData(authType: String, externalId: String): OptionT[F, String]` !

Reply:
The API takes userId as parameter and not externalId, so findCustomData is not a feasible operation to get the token for the userId.
The authservice does not provide a method to get the accounts for a user.
See the implementation of the findAccessToken method, it makes a db request to list accounts for the user.
Perhaps it would be better if there was a method in authService that lists the accounts for a userId or gets the account by authType and userId and the OAuth2Service makes use of that function and findCustomData.

Reply:
hmm.. yeah you are right, the `findCustomData` is not good for here. but then i think the `findAccessToken` should be renamed to `findProviderCustomDataByUserId` or something similar and the return type should be `OptionT[F, String]` just like the `findCustomData`.